### ISSUE TITLE
Multiple DICOM messages sent from Mirth (eg 130 slice cone beam CT), association closes and reopens for every message

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#1901]
My apologies if this is misdirected.

Please see posted forum issue: http://www.mirthcorp.com/community/forums/showthread.php?t=5897 for channel and output attachments

Setting up a DICOM to DICOM channel works fine if I send a single message. It grabs the message and forwards it on to the target AE. However, if I send a multi-message set, it sends it one message at a time! As in, it closes the association and re-opens after every message.
*** How can I get it so that the association stays open for as many messages are in the channel? ***

In case it is useful, I am using to send dcm4che's dcmsnd, and to receive both dcmrcv and MOSAIQ (Elekta/Impac radiation oncology software) Dcmwin.exe. MOSAIQ is the desired end result.

Multi-image sets require being sent as a single association for many Oncology and Med Imaging destinations (MOSAIQ, CHARM, Epic EMR, Pinnacle, Eclipse, etc etc) to be correctly imported into the patient image store.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-1930
Reporter: telemiel
Created: 2011-08-29T16:10:34.000-0700

**Comment by rbeckman-nextgen** [comment]
I believe this issue correlates directly with issue 1573: http://www.mirthcorp.com/community/issues/browse/MIRTH-1573 which has been around for over a year. I think the performance issue is because of the opening and closing of the dicom association.



Imported Comment. Original Details:
Author: amartin
Created: 2011-10-11T06:06:38.000-0700

**Comment by rbeckman-nextgen** [comment]
I can confirm that switching to a *nix server host (tried Debian, Solaris 11 Express, and Scientific Linux which is basically RHEL) makes no difference. Same versions of software and Java (carefully NOT using the openJDK on the Linux installations). Just in case it was the Windows networking. 

Not sure if that helps.

In all cases removing Mirthconnect from the equation lets the DICOM msgs flow smoothly in a single association.

Imported Comment. Original Details:
Author: telemiel
Created: 2011-10-11T17:56:11.000-0700

**Comment by rbeckman-nextgen** [comment]
I have just started testing MirthConnect as a DICOM tool and I also see each image sent as a separate association, significantly increasing TCP and DICOM overhead. I tried to send a large chest CT (6000 slices) through MirthConnect to get an indication of timing. After less than 600 images, it crashed. 

I can understand why MirthConnect sends each image in a single association - coming from a HL7 pedigree where this is the norm and where it is unknown from image to image whether more are to come. It would be good if the association with the destination could remain open till the inbound association closes or when there was no activity on the association for a set time (say 10 seconds). Alternatively, the images could be queued until the inbound association was closed, , then the queued images sent in a single association. This assumes an asynchronous channel. 

Just some thoughts after some initial testing.

Imported Comment. Original Details:
Author: garryc
Created: 2012-03-07T20:35:18.000-0800

**Comment by rbeckman-nextgen** [comment]
I agree Garry that a timeout is the best way to handle associations. A good majority of DICOM devices I work with have a user specified timeout period. That way if you are for example receiving some large mammo images over a slow WAN connection then you can increase your timeout to an appropriate value. Another example is ultrasound, if the study consist of multiple cine loops then the interval that images are received could vary significantly. Now the ultrasound example assumes that you would be sending directly from a modality to Mirth, which I see as a rare occasion.

I would really like to see some progress on this, and am willing to help in any way possible.  

Imported Comment. Original Details:
Author: amartin
Created: 2012-03-08T05:33:09.000-0800

**Comment by rbeckman-nextgen** [comment]
Does anybody know if there are any intentions to change how this is handled in Mirth today, or if there were any workaround solutions found to handle associations in Mirth?

Imported Comment. Original Details:
Author: brianda
Created: 2019-10-21T16:04:58.000-0700

**Comment by drseussofporn** [comment]
+1 

Has there been any updates to this? This is probably THE major issue surrounding the DICOM implementation of Mirth.

**Comment by tippenring** [comment]
We are also using Mirth for a portion of our DICOM workflow and would love to see a single association per study.
